net/dnscache/dnscache.go

```diff
@@ -580,6 +595,19 @@ func (dc *dialCall) raceDial(ctx context.Context, ips []netip.Addr) (net.Conn, e
 	}
 }
 
+func mergeSlices[T any](a, b []T) []T {
```

Comment:
should have a comment on what the expected output is. like merge based on what

Reply:
maybe `interleaveSlices`?

Reply:
Much better name :+1: